From 61c3a799fdf623a2acb14fe00ddc6048d30829e3 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 20 Nov 2023 11:43:57 +0100
Subject: [PATCH 0056/1695] core: mmu: fix nsec ddr discovery regarding
 MEM_AREA_RAM_NSEC

Memory registered as MEM_AREA_RAM_NSEC can be part of the non-secure
shared memory. This change fixes core_mmu_set_discovered_nsec_ddr()
to not check such memory area do not overlap with non-secure RAM.
This is no issue physical pages from MEM_AREA_RAM_NSEC memory are also
handled and mapped as dynamic non-secure memory by OP-TEE.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9ee8446297b4570b6753ea487545611ec4d9b265 core: mmu: fix nsec ddr discovery regarding MEM_AREA_RAM_NSEC]
Change-Id: Id54ee804f3fed4e9529bf9af5b5a8288c85d2d72
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
[etienne: cherry-picked from cbaf4c83e33dbf287d2c421f37f272426980f39e]
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/mm/core_mmu.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/mm/core_mmu.c b/core/mm/core_mmu.c
index 29310fe80..41fe9e4c1 100644
--- a/core/mm/core_mmu.c
+++ b/core/mm/core_mmu.c
@@ -471,6 +471,7 @@ void core_mmu_set_discovered_nsec_ddr(struct core_mmu_phys_mem *start,
 			break;
 		case MEM_AREA_EXT_DT:
 		case MEM_AREA_MANIFEST_DT:
+		case MEM_AREA_RAM_NSEC:
 		case MEM_AREA_RES_VASPACE:
 		case MEM_AREA_SHM_VASPACE:
 		case MEM_AREA_TS_VASPACE:
-- 
2.47.2

